### PR TITLE
Adding guidance on usage of $top for list messages and reorganizing examples

### PR DESCRIPTION
**Pull request by abheek-das**

### CODE REVIEW

File: api-reference/beta/api/user-list-messages.md [hunk 1]
```diff
@@ -15,7 +15,11 @@ Namespace: microsoft.graph
 
 Get the messages in the signed-in user's mailbox (including the Deleted Items and Clutter folders). 
 
-Depending on the page size and mailbox data, getting messages from a mailbox can incur multiple requests. The default page size is 10 messages. To get the next page of messages, simply apply the entire URL returned in `@odata.nextLink` to the next get-messages request. This URL includes any query parameters you may have specified in the initial request. 
+To improve the operation response time, use `$select` to specify the exact properties you need; see [example 1](#example-1-list-all-messages) below. Fine-tune the values for `$select` and `$top`, especially when you must use a larger page size, as returning a page with hundreds of messages each with a full response payload may trigger the [gateway timeout](/graph/errors#http-status-codes) (HTTP 504).
+
+Depending on the page size and mailbox data, getting messages from a mailbox can incur multiple requests. The default page size is 10 messages. Use `$top` to customize the page size, within the range of 1 and 1000.
```
Comment (angelgolfer-ms) [inline api-reference/beta/api/user-list-messages.md:20]: @abheek-das , can you please swap the order of the 2 paragraphs in line 18 and line 20?
I think it makes better sense to have the paragraph in line 20 first, which introduces the $top parameter.

File: api-reference/v1.0/api/user-list-messages.md [hunk 2]
```diff
@@ -13,7 +13,11 @@ Namespace: microsoft.graph
 
 Get the messages in the signed-in user's mailbox (including the Deleted Items and Clutter folders).
 
-Depending on the page size and mailbox data, getting messages from a mailbox can incur multiple requests. The default page size is 10 messages. To get the next page of messages, simply apply the entire URL returned in `@odata.nextLink` to the next get-messages request. This URL includes any query parameters you may have specified in the initial request. 
+To improve the operation response time, use `$select` to specify the exact properties you need; see [example 1](#example-1-list-all-messages) below. Fine-tune the values for `$select` and `$top`, especially when you must use a larger page size, as returning a page with hundreds of messages each with a full response payload may trigger the [gateway timeout](/graph/errors#http-status-codes) (HTTP 504).
+
+Depending on the page size and mailbox data, getting messages from a mailbox can incur multiple requests. The default page size is 10 messages. Use `$top` to customize the page size, within the range of 1 and 1000.
+
```
Comment (angelgolfer-ms) [inline api-reference/v1.0/api/user-list-messages.md:19]: Similarly, @abheek-das , can you please swap the order of the 2 paragraphs in line 16 and line 18?
I think it makes better sense to have the paragraph in line 18 first, which introduces the $top parameter.

File: api-reference/beta/api/user-list-messages.md (hunk 1 above)
Comment (abheek-das) [reply api-reference/beta/api/user-list-messages.md:18]: Fixed this.

File: concepts/query-parameters.md [hunk 3]
```diff
@@ -435,9 +435,9 @@ Use the `$top` query parameter to specify the page size of the result set.
 
 If more items remain in the result set, the response body will contain an `@odata.nextLink` parameter. This parameter contains a URL that you can use to get the next page of results. To learn more, see [Paging](./paging.md). 
 
-$top accepts a minimum value of 1 and a maximum value of 999 (inclusive).
```
Comment (angelgolfer-ms) [inline concepts/query-parameters.md:438]: @abheek-das , the next action item is not your making (but that of the Dev Relations team across the entire site), but contributors like yourself are sharing the load to fix the following issues in existing topics in Graph - if you don't mind, can you please do a search and replace in VS Code of the following, so as to get rid of the bunch of "suggestions" in your build results ("[Suggestion-preserve-view-not-set]") for the query-parameters.md topic:

(1) Search for:
/graph/api/resources/directoryobject?view=graph-rest-1.0
Replace by:
/graph/api/resources/directoryobject

(2) Search for:
/graph/api/resources/user?view=graph-rest-beta
Replace by:
/graph/api/resources/user?view=graph-rest-beta&preserve-view=true

The Dev Relations team documented this here: https://review.docs.microsoft.com/en-us/help/contribute/links-how-to?branch=master#links-to-specific-versions

At some point, they will turn these "suggestions" into build warnings.

Comment (abheek-das) [reply concepts/query-parameters.md:438]: @angelgolfer-ms Made the fixes as suggested!

**Review by angelgolfer-ms — APPROVED** [review]
Thanks for fixing those links!

**Pull request merged**